prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh

```diff
@@ -0,0 +1,159 @@
+
+if [[ $GARDENER_PROVIDER == "azure" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    ## TODO what should I put here? Is this a backend?
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it successful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set KYMA_SOURCE used by gardener::install_kyma
+# at the time of writing this comment, kyma-integration-gardener never sets BUILD_TYPE to "release"
+if [[ -n ${PULL_NUMBER} ]]; then
+    # In case of PR, operate on PR number
+    KYMA_SOURCE="PR-${PULL_NUMBER}"
+    export KYMA_SOURCE
+    # TODO maybe can be replaced with PULL_BASE_REF?
+elif [[ "${BUILD_TYPE}" == "release" ]]; then
+    readonly RELEASE_VERSION=$(cat "VERSION")
+    log::info "Reading release version from RELEASE_VERSION file, got: ${RELEASE_VERSION}"
+    KYMA_SOURCE="${RELEASE_VERSION}"
+    export KYMA_SOURCE
+else
+    # Otherwise (master), operate on triggering commit id
+    if [[ -n ${PULL_BASE_SHA} ]]; then
+        readonly COMMIT_ID="${PULL_BASE_SHA::8}"
+        KYMA_SOURCE="${COMMIT_ID}"
+        export KYMA_SOURCE
+    else
+        # periodic job, so default to main
+        KYMA_SOURCE="main"
+        export KYMA_SOURCE
+    fi
+fi
+
+# checks required vars and initializes gcloud/docker if necessary
+gardener::init
+
+# if MACHINE_TYPE is not set then use default one
+gardener::set_machine_type
+
+kyma::install_cli
+
+# currently only Azure generates overrides, but this may change in the future
+gardener::generate_overrides
+
+gardener::provision_cluster
+
+log::banner "Create EventMesh Secret"
+eventing::create_eventmesh_secret
+
+log::banner "Create Eventing Backend Secret"
+eventing::create_eventing_backend_secret
+
+# uses previously set KYMA_SOURCE
+if [[ "${KYMA_ALPHA}" == "true" ]]; then
+  kyma::alpha_deploy_kyma
+else
+  gardener::install_kyma
+fi
+
+# generate pod-security-policy list in json
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
+
+
+if [[ "${HIBERNATION_ENABLED}" == "true" ]]; then
+    gardener::hibernate_kyma
+    sleep 120
```

Comment:
How about making`sleep` configurable from job manifest?

Reply:
Added a todo for that.